src/box/lua/tuple.lua

```diff
@@ -79,8 +79,12 @@ local tuple_encode = function(tmpbuf, obj)
     elseif is_tuple(obj) then
         encode_r(tmpbuf, obj, 1)
     elseif type(obj) == "table" then
-        encode_array(tmpbuf, #obj)
-        for i = 1, #obj, 1 do
+        local obj_size = #obj
+        if obj_size == 0 and not rawequal(next(obj), nil) then
+            return box.error(box.error.TUPLE_NOT_ARRAY)  -- dictionary cannot represent a tuple
```

Comment:
The line is too long - should be < 80 characters. Please fix. Other than that looks good.